Mathlib/Data/Finset/Minimal.lean

```diff
@@ -0,0 +1,38 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Data.Fintype.Card
+import Mathlib.Order.Minimal
+
+/-!
+# Maximal upper and minimal lower bounds of elements in finsets
+
+Let a finset over a partially ordered type and an element in it be given, then
+
+* `Finset.exists_maximal_upper_bound` says that it has a maximal upper bound.
+* `Finset.exists_minimal_lower_bound` says that it has a minimal lower bound.
+-/
+
+namespace Finset
+
+variable {α : Type*} [PartialOrder α] {s : Finset α} {a : α}
+
+/-- Every element in a finset over a partially ordered type has a maximal upper bound. -/
+lemma exists_maximal_upper_bound (h : a ∈ s) : ∃ m ≥ a, Maximal (· ∈ s) m := by
```

Comment:
See also [Finite.exists_ge_minimal](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Order.html#Finite.exists_ge_minimal), which should be renamed to `Finite.exists_le_minimal`. Also, please add a version for `Set.Finite`.

Reply:
Done.